internal/clientcache/internal/daemon/options.go

```diff
@@ -40,6 +46,40 @@ func WithDebug(_ context.Context, debug bool) Option {
 	}
 }
 
+// withRefreshInterval provides an optional refresh interval.
+func withRefreshInterval(_ context.Context, d time.Duration) Option {
+	return func(o *options) error {
+		if d <= 0 {
+			return fmt.Errorf("provided refresh interval %q must be positive", d)
+		}
+		o.withRefreshInterval = d
+		return nil
+	}
+}
+
+// withFullFetchInterval provides an optional full fetch interval.
+func withFullFetchInterval(_ context.Context, d time.Duration) Option {
+	return func(o *options) error {
+		if d <= 0 {
+			return fmt.Errorf("provided full fetch interval %q must be positive", d)
+		}
+		o.withFullFetchInterval = d
+		return nil
+	}
+}
+
+// withIntervalRandomizationFactor provides an optional interval randomziation factor.
+func withIntervalRandomizationFactor(_ context.Context, f float64) Option {
```

Comment:
Is this currently used?

Reply:
I see it's supported in newRefreshTicker(...) but I didn't see any way for the
user to change/configure it.

Reply:
should we name it testWithIntervalRandomizationFactor(...) then?